src/transformers/models/idefics/modeling_idefics.py

```diff
@@ -140,47 +140,45 @@ class IdeficsCausalLMOutputWithPast(ModelOutput):
 
 
 def expand_inputs_for_generation(
-    input_ids,
+    input_ids=None,
     expand_size=1,
     is_encoder_decoder=False,
     attention_mask=None,
     encoder_outputs=None,
     **model_kwargs,
 ):
-    expanded_return_idx = (
-        torch.arange(input_ids.shape[0]).view(-1, 1).repeat(1, expand_size).view(-1).to(input_ids.device)
-    )
-    input_ids = input_ids.index_select(0, expanded_return_idx)
+    if input_ids is not None:
```

Comment:
For me seems like there is no difference and making it this way enables contrastive decoding in Idefics. Should we ask someone who made idefics to look?

Reply:
yes, the idefics model tests are passing, including slow